server/channels/store/sqlstore/user_store.go

```diff
@@ -263,6 +263,12 @@ func (us SqlUserStore) Update(rctx request.CTX, user *model.User, trustedUpdateD
 		}
 	}
 
+	// In the past, changing the email of a SSO user would mark the email as unverified.
+	// This is a lazy migration to fix broken records.
```

Comment:
Since it's a temporary fix, do you think adding a comment like: "This workaround was added in v11, and should be safe to remove in v13 or something" would be helpful?

Reply:
I don't think we should remove the workaround unless we add a proper migration to the app layer. There is no guarantee that the lazy migration will fix all broken users.